presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -713,6 +713,55 @@ public void testSchemaAuthorization()
         assertUpdate(admin, "DROP ROLE admin");
     }
 
+    @Test
+    public void testShowCreateSchema()
+    {
+        Session admin = Session.builder(getQueryRunner().getDefaultSession())
+                .setIdentity(Identity.forUser("hive")
+                        .withRole("hive", new SelectedRole(ROLE, Optional.of("admin")))
+                        .build())
+                .build();
+
+        Session user = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setSchema("test_show_create_schema")
+                .setIdentity(Identity.forUser("user").withPrincipal(getSession().getIdentity().getPrincipal()).build())
+                .build();
+
+        assertUpdate(admin, "CREATE ROLE authorized_users");
+        assertUpdate(admin, "GRANT authorized_users TO user");
+
+        assertUpdate(admin, "CREATE SCHEMA test_show_create_schema");
+        String createSchemaSql = format("" +
+                "CREATE SCHEMA %s.test_show_create_schema\n" +
+                "AUTHORIZATION USER hive\n" +
+                "WITH \\(\n" +
+                "   location = '.*test_show_create_schema'\n" +
```

Comment:
Hmm... I wouldn't know how to distinguish this. The location *is* set in the metadata, either a default or a given value. I'll dig into this.

Reply:
Schema is handled differently. The Hivemetastore sets the default, so it's always set. Without some bigger effort I cannot hide it if it is the Hive default.

Table location, on the other hand, is shown when the table's type is "external table", otherwise it isn't even set.

Reply:
Found a way by using the metastore's get_config_value. I also implemented that for the FileHiveMetastore by supporting the warehouse config there.

Reply:
Now of course I cannot test showing the location anymore. Adding this:
```
        // now test with a custom schema location
        File tempDir = createTempDir();
        assertUpdate(admin, format("CREATE SCHEMA test_schema_with_location WITH (location = '%s')", tempDir));

        createSchemaSql = format("" +
                "CREATE SCHEMA %s.test_schema_with_location\n" +
                "AUTHORIZATION USER hive\n" +
                "WITH \\(\n" +
                "   location = '%s'\n" +
                "\\)",
                getSession().getCatalog().get(),
                tempDir);

        actualResult = getOnlyElement(computeActual(admin, "SHOW CREATE SCHEMA test_schema_with_location").getOnlyColumnAsSet()).toString();
        assertEquals(actualResult, createSchemaSql);
        assertUpdate(user, "DROP SCHEMA test_schema_with_location");
```
Does not work, since the metastore used for tests does not allow setting schema,